# Credit Card

How to apply and how to issue credit card

### Author: WSN (Wall Street Newbie)
### Date: As of January 1, 2022

### Copyright: © Ailink Technology Corporation, All rights reserved 2022

A lot of material can be found on the [the points guy](https://thepointsguy.com/), this model only used to determine whether to apply for certain card or not

## Background
Credit card can be a **double-edge sword**, 
* On one hand, it can help you accure credit, which in turns help you **buy properties, build your business**. 
* On the other hand, it can drag you into debt quagmire.

So, the purpose of this modeling is to find a good credit card, just good for your buying power, and not drag you to over-purchase.

## Without SSN? 
Some credit card doesn't **require SSN**, for example, BOA's credit card.   
The good side of it was, you can apply it and use it for oldest credit line,   
and when you get SSN, you can link the card to it. 

## Overview of credit line
Credit line is composed of several parts:   
* **On-Time Payments**: a lot impact
* **Oldest Credit Line**: a lot impact
* **Credit Used**: moderate impact
* **Recent Inquiries**: small impact
* **New Accounts**: small impact
* **Available Credit**: small impact

More can be found on the app: Capital One CreditWise 

## Profit = Revenue - Cost <=> Basepoint
* Basepoint: 0

## Credit Card Model
the formula for geometric sequence
$$a_n = a_1 \times q^{n-1}$$

In [15]:
def base_point(new_account_bonus, referral_bonus=0, num_referral=0):
    # base point is calculated by how many points you earn with new account bonus
    # plus how many people you refer, and in total how many points you earn in referral
    return new_account_bonus + referral_bonus * num_referral

def geometric(a1, q, n):
    if a1 <= 0:
        print("a1 must be positive or 0")
        return 0
    if n < 1: 
        print("year must be positive")
        return 0
    if q == 0:
        return a1
    return a1 * (q ** (n - 1))


def calculate_expense(points, points_factor, annual_fee, year=10, annual_boost=0, new_account_bonus=0, referral_bonus=0, num_referral=0):
    '''
        points: the best category you will earn, i.e., the best power the card can give you
        for example: you name it gasoline card, since the card provides 4x points when you purchase gas.
        
        points_factor: how many real money does each points worth, usually 1 cents
        However, some rewards program makes each of your points 1.25x worth, i.e., 1 point 1.25 cents.
        We use cash back to demostrate. 
        
        annual_fee: annual fee
        
        
        annual_boost: sometimes the card provides annual boost on points, for example 10% increase
        year: a scale of time, use 5 year to illustrate
    '''
    base_points = base_point(new_account_bonus, referral_bonus, num_referral)
    total_return = geometric(base_points, annual_boost, year)
    
    # To determine whether to apply for a credit card, 
    # your category expense should be enough to cover annual fee
    
    total_cost = annual_fee * year // points_factor
    print("total cost: ", total_cost)
        
    print("total return: ", total_return)
    
    buffer = total_cost - total_return 
    
    annual_expense = (buffer // points) // year
    
    return annual_expense

## Amex Blue Cash Preferred 
**Points**: 6x at US supermarket
* US supermarket: wholefoods, safeway
* Not US supermarket: walgreen, costco  
[Check online](https://card.americanexpress.com/d/blue-cash-preferred-credit-card/)

**Points factor**: no more than 1 cents  
[Check online](https://www.cnbc.com/select/what-are-american-express-membership-rewards-points-worth/)  
**Annual Fee**: 1st year free, then 95  
**New account bonus**: 350 dollars = 35000 cents  

In [17]:
points = 6
points_factor = 0.01
year = 10
annual_fee = 95.0 * (year - 1) // year
annual_boost = 0
new_account_bonus = 35000

annual_expense = calculate_expense(points, points_factor, annual_fee, year, annual_boost, new_account_bonus)
annual_expense

total cost:  84999.0
total return:  35000


833.0

## Chase Sapphire Preferred 

## Citi Costco

That means, you need to at least spend 833 USD in US. supermarket annually to make the credit card worth. 

## Misc.